### ISSUE TITLE
"Update all" sometimes causes wrong versions to be applied

### ISSUE THREAD

**Issue by cachapa** [pszklarska/FlutterPubVersionChecker#64]
The "update all" functionality seems to cause wrong versions to be incorrectly set when packages have similar names.

Example:

```yaml
path_provider: ^2.0.8
provider: ^6.0.3
```

Results in:

```yaml
path_provider: ^6.0.5  # Broken, should be 2.1.1
provider: ^6.0.3       # Not updated, should be 6.0.5
```

It seems the problem is that the `provider` version has been applied to `path_provider`, presumably because both have the word `provider` in the package name.

**Comment by pszklarska** [comment]
Hey @cachapa thank you for this report! I'll look into this 👍
